### DISCUSSION TITLE
It is strongly requested to provide binary programs. Compilation is too troublesome.

### DISCUSSION THREAD

**Discussion by v2ado** [apache/kvrocks#1772] [Q&A] [answered]
It is strongly requested to provide binary programs. Compilation is too troublesome.

**Comment by PragmaTwice** [comment, marked as answer]
Hi @v2ado , it is a hard job for us to ship releases with pre-built binaries, so we decide not to maintain official binaries.

For the reason, you can refer to https://github.com/apache/kvrocks/discussions/1152#discussioncomment-4297513. Generally speaking, it is a very hard job to make binaries for every architechure, every linux distro and it versions, and for every GLIBC version (any difference in the enviroment may lead to a binary that cannot be used).

So if you do not want to build it by yourself, you can use our official Docker image, or some community maintained packages (e.g. in archlinux, you can use the [kvrocks](https://aur.archlinux.org/packages/kvrocks) package from AUR).

Besides, building it by yourself in your machine can let you achieve maximum performance, since you can compile it with native CPU instructions and more concrete ISA extensions.

BTW, if you encounter any difficulty or problem while building kvrocks, you can open an issue.
